File: 0905. Sort Array By Parity/Solution.java
class Solution {
    public int[] sortArrayByParity(int[] A) {
        int front = 0, rear = A.length - 1;
        while (front < rear) {
            if (A[front] % 2 == 1) {
                int tmp = A[front];
                A[front] = A[rear];
                A[rear--] = tmp;
            } else front++;
        }
        return A;
    }
}

File: 0905. Sort Array By Parity/Solution.py
from typing import List


class Solution:
    def sortArrayByParity(self, A: List[int]) -> List[int]:
        front, rear = 0, len(A) - 1
        while front < rear:
            if A[front] % 2:
                A[front], A[rear] = A[rear], A[front]
                rear -= 1
            else:
                front += 1
        return A
